src/stream.jl

```diff
@@ -229,50 +229,26 @@ end
     end
 end
 
-function Base.ismarked(stream::TranscodingStream)
+function Base.ismarked(stream::TranscodingStream)::Bool
     checkmode(stream)
-    return ismarked(stream.buffer1)
+    isopen(stream) && ismarked(stream.buffer1)
 end
 
-function Base.mark(stream::TranscodingStream)
-    checkmode(stream)
-    return mark!(stream.buffer1)
+function Base.mark(stream::TranscodingStream)::Int64
+    ready_to_read!(stream)
```

Comment:
This throws an error if `mark` is used on a stream in write, close, or panic modes.